### PR TITLE
add support for radial gradients

### PR DESCRIPTION
**Pull request by lsalzman**
This change is [Reviewable](https://reviewable.io/reviews/servo/webrender/734)

### CODE REVIEW

File: webrender/res/prim_shared.glsl [hunk 1]
```diff
@@ -31,6 +31,7 @@
 #define UV_PIXEL         uint(1)
 
 #define MAX_STOPS_PER_ANGLE_GRADIENT 8
+#define MAX_STOPS_PER_RADIAL_GRADIANT 8
```
Comment (glennw) [inline webrender/res/prim_shared.glsl]: nit: gradiant -> gradient

File: webrender/src/renderer.rs [hunk 2]
```diff
@@ -473,6 +475,10 @@ impl Renderer {
                                                      &mut device,
                                                      &[],
                                                      options.precache_shaders);
+        let ps_radial_gradient = PrimitiveShader::new("ps_radial_gradient",
+                                                     &mut device,
```
Comment (glennw) [inline webrender/src/renderer.rs]: nit: indentation not quite right

File: webrender/res/ps_radial_gradient.fs.glsl [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        if (vStartRadius + rd * t0 >= 0.0) {
+            x = t0;
+        } else if (vStartRadius + rd * t1 >= 0.0) {
+            x = t1;
+        } else {
+            discard;
+        }
+    }
+
+    oFragColor = mix(vColors[0],
```
Comment (kvark) [inline webrender/res/ps_radial_gradient.fs.glsl]: Can the radial gradient be masked? I think we are missing a mask fetch here.

**Comment by lsalzman** [comment]
Based the shader off the angle gradient shader, which didn't do it either.
So presumably if radial gradients are missing that, then so are angle
gradients...

**Pull request merged**